File: frontend/src/types/repository.ts
export interface Repository {
  id: string;
  project_id: string;
  name: string;
  clone_url?: string;
  storage_path?: string;
  status: 'pending' | 'scanning' | 'scanned' | 'error';
  total_files: number;
  created_at: string;
}

export interface FileTreeNode {
  name: string;
  type: 'directory' | 'file';
  path?: string;
  size?: number;
  extension?: string;
  children?: FileTreeNode[];
}

export interface FileTreeResponse {
  repository_id: string;
  repository_name: string;
  total_files: number;
  file_tree: FileTreeNode;
}

export interface FileContentResponse {
  repository_id: string;
  file_path: string;
  content: string;
  size_bytes: number;
  file_type: string;
}

File: backend/app/schemas/repository.py
from datetime import datetime
from typing import Optional, Dict, Any, List
from pydantic import BaseModel, Field, ConfigDict


class GitHubCloneInput(BaseModel):
    repo_url: str = Field(..., description="Public GitHub repository clone URL (e.g. https://github.com/fastapi/fastapi.git)")
    name: Optional[str] = Field(None, description="Custom display name for the repository")


class CodeFileSummary(BaseModel):
    id: str
    file_path: str
    file_type: str
    size_bytes: int

    model_config = ConfigDict(from_attributes=True)


class RepositoryResponse(BaseModel):
    id: str
    project_id: str
    name: str
    clone_url: Optional[str] = None
    storage_path: Optional[str] = None
    status: str
    total_files: int
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)


class FileTreeResponse(BaseModel):
    repository_id: str
    repository_name: str
    total_files: int
    file_tree: Dict[str, Any]


class FileContentResponse(BaseModel):
    repository_id: str
    file_path: str
    content: str
    size_bytes: int
    file_type: str
